solr/core/src/java/org/apache/solr/handler/admin/ClusterStatus.java

```diff
@@ -183,10 +181,12 @@ private void fetchClusterStatusForCollOrAlias(
     String routeKey = solrParams.get(ShardParams._ROUTE_);
     String shard = solrParams.get(ZkStateReader.SHARD_ID_PROP);
 
-    Set<String> requestedShards = new HashSet<>();
+    Set<String> requestedShards;
     if (shard != null) {
       String[] paramShards = shard.split(",");
-      requestedShards.addAll(Arrays.asList(paramShards));
+      requestedShards = Set.of(paramShards);
+    } else {
+      requestedShards = Set.of();
     }
```

Comment:
I bet this whole thing could be one line using ternary operator and no line wrap. 
Also, if "shard" isn't specified, I think it's clearer to have requestedShards as null to indicate there was no request for any shards.